.github/workflows/trufflehog.yml

```diff
@@ -0,0 +1,21 @@
+on:
+  push:
+
+name: Secret Leaks
+
+permissions:
+  contents: read
+  id-token: write
+  issues: write
+  pull-requests: write
```

Comment:
Thank you for this!
I'll add a pre-commit as well (Ideally we want to catch these things before they're even pushed)

Do you know why we need those write permissions though?

Reply:
Here's a link to the pre-commit hook if you haven't seen it already: git@github.com:huggingface/notebooks.git.

I'm checking for the permissions, afaict we could actually remove them. I copied the CI from an internal repo without checking, should've been more thorough here. I think it may be a leftover from another CI pipeline.

Reply:
Thanks.
Yes, I've seen their pre-commit example but I really don't want to use it as `- repo: local` because it assumes the dev installs it on its end, and I don't want our pre-commits install to be an additional pain point for our contributors.

Ideally I'd like to use something like this:
```yaml
- repo: https://github.com/trufflesecurity/trufflehog
    rev: v3.78.0
    hooks:
      - id: trufflehog
```
But it's unclear if this works as intended.

Reply:
In the example:
```yml
repos:
  - repo: local
    hooks:
      - id: trufflehog
        name: TruffleHog
        description: Detect secrets in your data.
        entry: bash -c 'trufflehog git file://. --since-commit HEAD --only-verified --fail'
        # For running trufflehog in docker, use the following entry instead:
        # entry: bash -c 'docker run --rm -v "$(pwd):/workdir" -i --rm trufflesecurity/trufflehog:latest git file:///workdir --since-commit HEAD --only-verified --fail'
        language: system
        stages: ["commit", "push"]
```

They provide the command for running trufflehog in docker, does that not work for you?

Reply:
That still assumes to have docker installed.
For instance, we develop on the cluster and we don't have access to `docker run`.

Reply:
Anyway, I think we can merge this in the meantime. Thanks ;)

Reply:
I understand where you're coming from, I wonder if it's possible to run a preinstall step when installing the pre commit hook. You could download the trufflehog binary then.
Otherwise, `make install`? 😅